packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -514,6 +548,35 @@ class Scaffold extends StatefulWidget {
     );
   }
 
+  /// Returns a [ValueNotifier] for the [ScaffoldGeometry] for the closest
+  /// [Scaffold] ancestor of the given context.
+  ///
+  /// Value notifications are sent during the layout pass.
```

Comment:
docs should talk about how this object must be discarded any time the given context has its dependencies change, notably, for the context from a StatefulWidget, whenever the corresponding State's didChangeDepndencies is fired.

Reply:
Added some details to the doc